### PR TITLE
MyGet Transistion

### PR DESCRIPTION
**Pull request by markwilkie**
Now that our primary dependency is on Azure DevOps feeds for .NET Core 3.x, it's time to finish the work so that we can completely discontinue our use of MyGet.  The main business driver is senior management's strong desire to take our mission critical dependencies on end points we own, which in this case is AzDO over MyGet.

As such, this is the initial plan (approach) to make this happen.  Your input is really important as this will most likely be pretty noisy....

Mark

### CODE REVIEW

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+5. Finish cleaning up (April)
+
+## Details
+
+### Investigation and Socialization
+- The scope include all feeds on dotnet.myget.org which includes servicing.
+- There will need to be communication to our community about this change.
+- This migration will be disruptive, so setting clear expectations, and giving us plenty of time to adjust will be key to our success.
+- There are a variety of partner teams which depend on MyGet today to get updates to .NET Core.  These consumers need to be identified as part of the investigation/socialization effort so that they can move their dependencies.
+- Further investigation is needed to determine the best approach for other types of packages, e.g. VSIX.  Universal packages might work...
```
Comment (cartermp) [inline Documentation/MigrationPlan/MyGetFeeds.md]: Drafting this to record that we produce two VSIXs in our feeds for nightly builds of the F# compiler and tools for VS

**Comment by vatsan-madhavan** [comment]
Added @rladuca to review (I’m on vacation)

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+- MyGet is backed by our own Azure storage location.  The idea is to bulk load a new "legacy" feed in Azure DevOps with all of our existing packages in MyGet.
+- This should allow our servicing builds to simply add a new feed to their nuget.config.
+- It's very important however, that no new packages be published to this legacy feed.  The eventual idea is that we can simply "turn off" the legacy feed once the usage is either gone or minimal.
+
+### Turn off MyGet (and clean up)
+- The two main factors which should make this transition reasonable are:
+  - We don't have that many dependencies on MyGet left in our 3.x builds (relatively speaking)
+  - There will be a "legacy" feed for existing packages.
+  - The migration is being staged over time
+- It is understood that this transition wil be noisy and the engineering servicing team is committed to doing whatever is necessary to help out where and as needed.
```
Comment (dagood) [inline Documentation/MigrationPlan/MyGetFeeds.md]: This is the correction @jcagme was making--subtle when there's no highlighting but there was a difference. 🙂 

```suggestion
- It is understood that this transition will be noisy and the engineering servicing team is committed to doing whatever is necessary to help out where and as needed. 
```

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+2. Make all MyGet feeds read-only (first part of Feb)
+3. Bulk load a new ‘legacy’ feed in Azure DevOps from the existing MyGet feeds (end of Feb)
+4. Turn off MyGet (first week of March)
+5. Finish cleaning up (April)
+
+## Details
+
+### Investigation and Socialization
+- The scope include all feeds on dotnet.myget.org which includes servicing.
+- There will need to be communication to our community about this change.
```
Comment (jaredpar) [inline Documentation/MigrationPlan/MyGetFeeds.md]:
```suggestion
- Will will need to communicate to our community about this change.
```

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+- This migration will be disruptive, so setting clear expectations, and giving us plenty of time to adjust will be key to our success.
+- There are a variety of partner teams which depend on MyGet today to get updates to .NET Core.  These consumers need to be identified as part of the investigation/socialization effort so that they can move their dependencies.
+- Further investigation is needed to determine the best approach for other types of packages, e.g. VSIX.  Universal packages might work...
+
+
+### Make MyGet Feeds Read-Only
+- This stops the publishing of new packages which gives a stable "snapshot" to migrate.
```
Comment (jaredpar) [inline Documentation/MigrationPlan/MyGetFeeds.md]:
```suggestion
- This prevents the publishing of new packages which gives a stable "snapshot" to migrate.
```

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+- Before this happens, it's very important that partner teams are aware that new packages will only be published to the Azure DevOps feeds.
```
Comment (jaredpar) [inline Documentation/MigrationPlan/MyGetFeeds.md]:
```suggestion
- Before this happens, it's very important that partner teams are aware that new packages can only be published to the Azure DevOps feeds.  
```

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+2. Make all MyGet feeds read-only (first part of Feb)
+3. Bulk load a new ‘legacy’ feed in Azure DevOps from the existing MyGet feeds (end of Feb)
+4. Turn off MyGet (first week of March)
+5. Finish cleaning up (April)
+
+## Details
+
+### Investigation and Socialization
+- The scope includes all feeds on dotnet.myget.org which includes servicing.
+- Will will need to communicate to our community about this change.
```
Comment (dagood) [inline Documentation/MigrationPlan/MyGetFeeds.md]: I'm sure Will can handle it, but is this meant to be:

```suggestion
- We will need to communicate to our community about this change.
```

Comment (markwilkie) [reply Documentation/MigrationPlan/MyGetFeeds.md]: LOL, yes - thanks @dagood

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+|	dotnet-core	|	various	|	1 day ago	|	NuGet (only Nuget.Build.Tasks.Pack), UWP6.0, Framework?	|	Some System.ServiceModel.Duplex 4.5.4 packges)
+|	dotnet-core-dev-api	|	runtime	|	3 years ago	|		|	
+|	dotnet-core-dev-defaultintf 	|	runtime	|	3 years ago	|		|	
+|	dotnet-core-dev-eng	|	runtime	|	2 years ago	|		|	
+|	dotnet-core-rel	|	runtime	|	4 years ago	|		|	
+|	dotnet-core-test	|	runtime	|	10 months ago	|		|	
+|	dotnet-coreclr	|	runtime	|	5 years ago	|		|	
+|	dotnet-corefxlab	|	corefxlab	|	5 days ago	|	Corefxlab folks	|	
+|	dotnet-stage	|	aspnet	|	3 years ago	|		|	
+|	msbuild	|	msbuild	|	5 days ago	|	msbuild	|
```
Comment (rainersigwald) [inline Documentation/MigrationPlan/MyGetFeeds.md]: We disabled publishing by default in ~August. AFAIK it's not needed anywhere now.

This recent publish was a manual intervention from @Forgind. Nathan, was that required or accidental?

**Comment by markwilkie** [comment]
Please review and get any additional comments you might have as I'd like to post this as our plan of record by early next week.

File: Documentation/MigrationPlan/MyGetFeeds.md (hunk 7 above)
Comment (Forgind) [reply Documentation/MigrationPlan/MyGetFeeds.md]: Not required

File: Documentation/MigrationPlan/MyGetFeeds.md (hunk 1 above)
Comment (markwilkie) [reply Documentation/MigrationPlan/MyGetFeeds.md]: Thanks - be sure and make sure you're taken care of once we start the migration.

File: Documentation/MigrationPlan/MyGetFeeds.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
+# (DRAFT) Plan to discontinue our use of MyGet
+
+## Summary
+Now that our primary dependency is on Azure DevOps feeds for .NET Core 3.x, it's time to finish the work so that we can completely discontinue our use of MyGet.  The main business driver is senior management's strong desire to take our mission critical dependencies on end points we own, which in this case is AzDO over MyGet.
+
+## Scope and Impact
+- We will be shutting down dotnet.myget.org.
+- Based on a quick audit, the following are the primary folks still pushing.  (e.g. impacted) --- 
+Fsharp •	Roslyn •	Tomas •	UWP •	Corefxlab •	NuGet •	MSBuild
```
[the remark this answers is not in the document]
Comment (markwilkie) [reply Documentation/MigrationPlan/MyGetFeeds.md]: fixed

File: Documentation/MigrationPlan/MyGetFeeds.md (hunk 7 above)
Comment (markwilkie) [reply Documentation/MigrationPlan/MyGetFeeds.md]: Thanks!

**Pull request merged**